File: web2-project/src/app/model/work-account.ts
import { Incident } from './incident';
import { Address } from './address';
import { Device } from './device';
import { User } from './user';

export enum Type{
    PlannedWork,
    UnplannedWork
}

export enum Status{
    Draft,
    Approved,
    Denied,
    Canceled
}

export class WorkAccount {
    constructor(public workAccountID:string, public type: Type, public status: Status, public incident: Incident, public address: Address, public startDate: Date,
 public endDate: Date, public created: Date, public purpose: string, public notes: string, public urgentWork : boolean, public company: string,
 public phoneNumber: number, public creationDate: Date, public equipment : Device, public workAccountStatusHistory : WorkAccountStatusHistory){}
}

export class WorkAccountStatusHistory{
    public date: Date;
    public changedBy: User;
    public status: Status;

    constructor(date: Date, user: User, status: Status){
        this.date = date;
        this.changedBy = user;
        this.status = status;
    }
}

File: Web2BackEnd/Web2BackEnd/Models/WorkAccount.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace Web2BackEnd.Models
{
    public class WorkAccount
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int WorkAccountID { get; set; }

        [Required]
        public TypeWorkAccount Type { get; set; }
        [Required]
        public StatusWorkAccount Status { get; set; }
        [Required]
        public int IncidentID { get; set; }
        [Required]
        public Address Address { get; set; }
        [Required]
        public DateTime StartDate { get; set; }
        [Required]
        public DateTime EndDate { get; set; }
        [Required]
        public string Created { get; set; }
        [Required]
        public DateTime CreationDate { get; set; }
        [Required]
        public string Purpose { get; set; }
        [Required]
        public string Notes { get; set; }
        [Required]
        public bool UrgentWork { get; set; }
        [Required]
        public string Company { get; set; }
        [Required]
        public string PhoneNumber { get; set; }

        public WorkAccountStatusHistory WorkAccountStatusHistory { get; set; }



    }

    public enum TypeWorkAccount
    {
        PlannedWork,
        UnplannedWork
    }

    public enum StatusWorkAccount
    {
        Draft,
        Approved,
        Denied,
        Canceled
    }
}
